### PR TITLE
[Intl] Update ICU data to 57.1

### PR DESCRIPTION
**Pull request by jakzal**
| Q | A |
| --- | --- |
| Branch? | 2.7 |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #19315 |
| License | MIT |
| Doc PR | - |

I think the only thing that makes sense with ICU is to always be on the latest available version.

### CODE REVIEW

**Comment by HeahDude** [comment]
👍 closes https://github.com/symfony/symfony/issues/19315.

File: src/Symfony/Component/Intl/Resources/data/currencies/en_150.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+    "Version": "2.1.19.76",
+    "Names": {
+        "EUR": [
+            "€",
+            "Euro",
+            {}
```
Comment (stof) [inline src/Symfony/Component/Intl/Resources/data/currencies/en_150.json]: is it expected to have an empty object here alongside strings ?

Comment (jakzal) [inline src/Symfony/Component/Intl/Resources/data/currencies/en_150.json]: This is generated. I'll double check if there's something wrong on our side.

Comment (stof) [inline src/Symfony/Component/Intl/Resources/data/currencies/en_150.json]: I know this is generated, but this means our APIs exposing currency name will return invalid data (their contract is to return an array of strings).

If there is a bug in the original files, this should be reported to the ICU project, and we should filter out invalid values in our generator script

Comment (jakzal) [inline src/Symfony/Component/Intl/Resources/data/currencies/en_150.json]: We don't seem to be using this value. There are occurrences of this third element in previous versions of icu data (see [here](https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Intl/Resources/data/currencies/ca.json#L356-L360]) and [here](https://github.com/symfony/symfony/blob/65a8b9fc93cc05857f7865806110030980c1c8c2/src/Symfony/Component/Intl/Resources/data/currencies/ca.json#L356-L360)).

The original ICU [file stores](http://source.icu-project.org/repos/icu/icu/tags/release-57-1/source/data/curr/en_150.txt) some additional formatting definitions here:

```
en_150{
    %%Parent{"en_001"}
    Currencies{
        EUR{
            "€",
            "Euro",
            {
                "¤#,##0.00",
                ",",
                ".",
            }
        }
    }
    Version{"2.1.19.76"}
}
```

We seem to be removing it.

**Comment by javiereguiluz** [comment]
👍 

@jakzal is this still WIP or is it ready to be merged? Thanks!

**Comment by jakzal** [comment]
@javiereguiluz this is indeed ready now. I was looking into updating ICU data on travis so we could run related tests, but I can do it as a separate PR (tests are not run for the  current version either).

**Comment by nicolas-grekas** [comment]
:+1:

**Comment by fabpot** [comment]
Thank you @jakzal.

**Pull request closed without merging**